From ceba2b67614c8393a8f11f5d8a913a5d13c35836 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Verschelde?= <rverschelde@gmail.com>
Date: Tue, 25 Feb 2020 13:20:04 +0100
Subject: [PATCH] Expression: Fix parsing integers as 32-bit

---
 core/math/expression.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/math/expression.cpp b/core/math/expression.cpp
index 11309355670..2728da616e1 100644
--- a/core/math/expression.cpp
+++ b/core/math/expression.cpp
@@ -1161,7 +1161,7 @@ Error Expression::_get_token(Token &r_token) {
 					if (is_float)
 						r_token.value = num.to_double();
 					else
-						r_token.value = num.to_int();
+						r_token.value = num.to_int64();
 					return OK;
 
 				} else if ((cchar >= 'A' && cchar <= 'Z') || (cchar >= 'a' && cchar <= 'z') || cchar == '_') {

