From 55b478afd912d4b6788f736efa87a33777aba5a4 Mon Sep 17 00:00:00 2001
From: JFonS <joan.fonssanchez@gmail.com>
Date: Tue, 10 Mar 2020 17:56:46 +0100
Subject: [PATCH] Update all editor viewports after settings changes

---
 editor/plugins/spatial_editor_plugin.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/editor/plugins/spatial_editor_plugin.cpp b/editor/plugins/spatial_editor_plugin.cpp
index 647d64c6274..624b14c43c4 100644
--- a/editor/plugins/spatial_editor_plugin.cpp
+++ b/editor/plugins/spatial_editor_plugin.cpp
@@ -6241,6 +6241,10 @@ SpatialEditor::SpatialEditor(EditorNode *p_editor) {
 	settings_zfar->set_value(EDITOR_DEF("editors/3d/default_z_far", 1500));
 	settings_vbc->add_margin_child(TTR("View Z-Far:"), settings_zfar);
 
+	for (uint32_t i = 0; i < VIEWPORTS_COUNT; ++i) {
+		settings_dialog->connect("confirmed", callable_mp(viewports[i], &SpatialEditorViewport::_update_camera), varray(0.0));
+	}
+
 	/* XFORM DIALOG */
 
 	xform_dialog = memnew(ConfirmationDialog);

