From 46bd01efa3a6a95ba662f999a1aa1204f94d3d1f Mon Sep 17 00:00:00 2001
From: Phischermen <kevfischermen@gmail.com>
Date: Fri, 13 Mar 2020 21:36:41 -0700
Subject: [PATCH] Stop ViewportTexture error spam

---
 scene/main/viewport.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scene/main/viewport.cpp b/scene/main/viewport.cpp
index 563fcb99617..670fb49fcd2 100644
--- a/scene/main/viewport.cpp
+++ b/scene/main/viewport.cpp
@@ -87,7 +87,8 @@ void ViewportTexture::set_viewport_path_in_scene(const NodePath &p_path) {
 	path = p_path;
 
 	if (get_local_scene()) {
-		setup_local_to_scene();
+		//Refresh texture. Call deferred because node path could be invalid during instantiation
+		call_deferred("setup_local_to_scene"); 
 	}
 }
 

